main.go

```diff
@@ -38,8 +42,13 @@ func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
 func corsMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		origin := r.Header.Get("Origin")
```

Comment:
Using an environment variable for CORS origins allows for flexibility across different environments (dev, staging, prod) without requiring code changes. This is a security best practice to avoid hardcoding trusted origins.